web/controllers/pair_controller.ex

```diff
@@ -4,64 +4,71 @@ defmodule Pairmotron.PairController do
   """
   use Pairmotron.Web, :controller
 
-  import Pairmotron.ControllerHelpers
   alias Pairmotron.PairMaker
 
-  @spec index(%Plug.Conn{}, map()) :: %Plug.Conn{}
+  @spec index(Plug.Conn.t, map()) :: Plug.Conn.t
   def index(conn, _params) do
     {year, week} = Timex.iso_week(Timex.today)
     user = conn.assigns[:current_user]
-    {conn, groups_and_pairs} = conn
-      |> assign_current_user_pair_retro_for_week(year, week)
-      |> fetch_pairs(year, week, user)
-    render conn, "index.html", year: year, week: week, groups_and_pairs: groups_and_pairs,
-      start_date: Timex.from_iso_triplet({year, week, 1}),
-      stop_date: Timex.from_iso_triplet({year, week, 7})
+    {groups_and_pairs, messages} = user |> fetch_groups_and_pairs(year, week)
+
+    conn
+    |> flash_messages(messages)
+    |> render_index(year, week, groups_and_pairs)
   end
 
-  @spec show(%Plug.Conn{}, map()) :: %Plug.Conn{}
+  @spec show(Plug.Conn.t, map()) :: Plug.Conn.t
   def show(conn, %{"year" => y, "week" => w}) do
     {year, _} = y |> Integer.parse
     {week, _} = w |> Integer.parse
     user = conn.assigns[:current_user]
-    {conn, groups_and_pairs} = conn
-      |> assign_current_user_pair_retro_for_week(year, week)
-      |> fetch_pairs(year, week, user)
-    render conn, "index.html", year: year, week: week, groups_and_pairs: groups_and_pairs,
-      start_date: Timex.from_iso_triplet({year, week, 1}),
-      stop_date: Timex.from_iso_triplet({year, week, 7})
+    {groups_and_pairs, messages} = user |> fetch_groups_and_pairs(year, week)
+
+    conn
+    |> flash_messages(messages)
+    |> render_index(year, week, groups_and_pairs)
+  end
+
+  @spec render_index(Plug.Conn.t, integer(), 1..53, [{Types.group, Types.pair}]) :: Plug.Conn.t
+  defp render_index(conn, year, week, groups_and_pairs) do
+    render(conn, "index.html", year: year, week: week, groups_and_pairs: groups_and_pairs,
+              start_date: Timex.from_iso_triplet({year, week, 1}),
+              stop_date: Timex.from_iso_triplet({year, week, 7}))
   end
 
-  @spec fetch_pairs(%Plug.Conn{}, integer(), 1..53, Types.user) :: {%Plug.Conn{}, [{Types.group, Types.pair}]}
-  defp fetch_pairs(conn, year, week, user) do
-    user = user
-      |> Repo.preload(:groups)
-    {conn, pairs} = user
-      |> pairs_for_user_groups(conn, year, week)
-    {conn, Enum.zip(user.groups, pairs)}
+  @spec fetch_groups_and_pairs(Types.user, integer(), 1..53) :: {[{Types.group, Types.pair}], [String.t]}
+  defp fetch_groups_and_pairs(user, year, week) do
+    user = user |> Repo.preload(:groups)
+    {pairs, messages} = pairs_for_user_groups(user, year, week)
+    {Enum.zip(user.groups, pairs), messages}
```

Comment:
So I think this line is dangerous. It usually works, but there's no reason to assume that there will always be one pair per group the user is in. It's not crazy to think that there could be something down the line which allows multiple concurrent pairs for one person. Also, as soon as group-specific activity is enabled, this could easily match up pairs with incorrect groups.

I'm not going to fix it in this PR, but I probably will for the PR that fixes #126